src/RadixSortLSD.chpl

```diff
@@ -78,28 +79,30 @@ module RadixSortLSD
             // count digits
             coforall loc in Locales {
                 on loc {
+                    // allocate counts
+                    var tasksBucketCounts: [Tasks] [0..#numBuckets] int;
                     coforall task in Tasks {
-                        // bucket domain
-                        var bD = {0..#numBuckets};
-                        // allocate counts
-                        var taskBucketCounts: [bD] int;
+                        ref taskBucketCounts = tasksBucketCounts[task];
                         // get local domain's indices
                         var lD = aD.localSubdomain();
                         // calc task's indices from local domain's indices
                         var tD = calcBlock(task, lD.low, lD.high);
-                        try! rsLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
-                                   "locid: %t task: %t tD: %t".format(loc.id,task,tD));
                         // count digits in this task's part of the array
                         for i in tD {
                             const key = comparator.key(temp.localAccess[i]);
                             var bucket = getDigit(key, rshift, last, negs); // calc bucket from key
                             taskBucketCounts[bucket] += 1;
                         }
-                        // write counts in to global counts in transposed order
+                    }//coforall task
+                    // write counts in to global counts in transposed order
+                    coforall tid in Tasks {
                         var aggregator = newDstAggregator(int);
-                        for bucket in bD {
-                            aggregator.copy(globalCounts[calcGlobalIndex(bucket, loc.id, task)], 
-                                                         taskBucketCounts[bucket]);
+                        for task in Tasks {
+                            ref taskBucketCounts = tasksBucketCounts[task];
+                            for bucket in chunk(0..#numBuckets, numTasks, tid) {
+                                aggregator.copy(globalCounts[calcGlobalIndex(bucket, loc.id, task)],
```

Comment:
this seems ok to me. although my brain goes a little numb following the index reordering.

Reply:
The indexing is actually the same as before. The change here is that previously each task would scatter/gather all it's own buckets and now tasks gather/scatter chunks of all the buckets on a node.